### PR TITLE
s3: set ContentType when putting object

### PR DESCRIPTION
**Pull request by 201341**
Signed-off-by: swj <1186093704@qq.com>

### CODE REVIEW

**Comment by davies** [comment]
This is useful for `juicefs sync`, can you add it for all object storage, or most of them?

**Comment by 201341** [comment]
> This is useful for `juicefs sync`, can you add it for all object storage, or most of them?

ok。

File: pkg/object/ibmcos.go [hunk 1]
```diff
@@ -79,10 +81,15 @@ func (s *ibmcos) Put(key string, in io.Reader) error {
 		}
 		body = bytes.NewReader(data)
 	}
+	mimeType := mime.TypeByExtension(path.Ext(key))
+	if !strings.ContainsRune(mimeType, '/') {
```
Comment (davies) [inline pkg/object/ibmcos.go:85]: we can have a function there these lines:  guessMimeType

Comment (201341) [reply pkg/object/ibmcos.go:85]: which file should I put this function on? utils.go or object_storage.go?

Comment (davies) [reply pkg/object/ibmcos.go:85]: utils.go is good

**Pull request merged**